streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNodeRecordDeserializer.java

```diff
@@ -54,4 +58,28 @@
                                     rawRecord.serializedValueSize(), key, value);
 
     }
+
+    public ConsumerRecord<Object, Object> tryDeserialize(final ProcessorContext processorContext,
+                                                         ConsumerRecord<byte[], byte[]> rawRecord) {
+
+        if (deserializationExceptionHandler == null) {
```

Comment:
Can we assume and/or ensure that this is non-null? Then we won't need this check here and can just have a single path through this method. I think we default to `LogAndFail...` if nothing is set - right?